content/docs/iac/adopting-pulumi/migrating-to-pulumi/from-cdk.md

```diff
@@ -0,0 +1,469 @@
+
+For instance, let's say your infrastructure team has provisioned your network infrastructure using CDK and you need to use the Subnet ID to provision something new from your Pulumi program. One approach is to hardcode that ID but this is brittle and, if it ever changes, you'd need to go and manually update the hardcoded value.
+
+Instead, you can look up that stack by name and use one of its output values. The following example reads an AWS CDK stack named `my-network-stack` and then uses the exported `SubnetId` value to provision a brand new EC2 instance that runs in that subnet:
+
+```ts
+import * as aws from "@pulumi/aws";
+
+const network = aws.cloudformation.getStackOutput({
+    name: "my-network-stack",
+});
+
+const subnetId = network.outputs["SubnetId"];
+
+const web = new aws.ec2.Instance("web", {
+    ami: "ami-0adc0e3ef2558cb1f", // us-west-2 AMI
+    instanceType: "t3.micro",
+    subnetId: subnetId,
+});
+```
+
+All we need to do is run `pulumi up` and the Pulumi runtime will know how to query the CDK stack and retrieve its output values. In this case, the CDK stack is treated entirely as read-only, and Pulumi will never attempt to modify it or any resources managed by it.
+
+> Although we've hard-coded the AWS CDK stack name here &mdash; `"my-network-stack"` &mdash; it's common to dynamically compute a name using unique per-stack information, like the stack name, AWS region, or other configuration variables.
+
+## Converting CDK Applications
+
+Let's say you want to migrate from AWS CDK to Pulumi and that simply co-existing side-by-side as shown above isn't sufficient. In this case you can convert your AWS CDK application to a Pulumi CDK application using the [Pulumi CDK Adapter](https://github.com/pulumi/pulumi-cdk).
+
+Using the below example as a starting point, we will convert this CDK application to use Pulumi CDK.
+
+```ts
+import * as cdk from 'aws-cdk-lib/core';
+import { Construct } from 'constructs';
+import * as s3 from 'aws-cdk-lib/aws-s3';
+
+class TestStack extends cdk.Stack {
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+
+    new s3.Bucket(this, 'Bucket');
+
+  }
+}
+
+const app = new cdk.App();
+new TestStack(app, 'import-test');
+```
+
+### Install Pulumi Dependencies
+
+The Pulumi CDK Adapter requires the installation of the following npm packages:
+
+* [@pulumi/aws](https://www.npmjs.com/package/@pulumi/aws)
+* [@pulumi/aws-native](https://www.npmjs.com/package/@pulumi/aws-native)
+* [@pulumi/docker-build](https://www.npmjs.com/package/@pulumi/docker-build)
+* [@pulumi/pulumi](https://www.npmjs.com/package/@pulumi/pulumi)
+
+```console
+npm install @pulumi/aws @pulumi/aws-native @pulumi/docker-build @pulumi/pulumi
```

Comment:
Is there not a way to install @pulumi/cdk and get them pulled in transitively?

Reply:
`npm` should automatically install peerDependencies, but sometimes there can be issues and I don't think `yarn` does. I think it's safer to just have the full list of dependencies the user can install.